### PR TITLE
🐛 Fix validation of DIRECTOR_V2_SERVICES_CUSTOM_CONSTRAINTS: Make it conform to docker specs

### PR DESCRIPTION
**Pull request by mrnicegyu11**
## What do these changes do?

Allow underscore in DIRECTOR_V2_SERVICES_CUSTOM_CONSTRAINTS in regex validation during container start

Before:
`node.labels.standard_worker==true` --> invalid
`node.labels.standardworker==true` --> valid

Now:
`node.labels.standard_worker==true` --> valid
`node.labels.standardworker==true` --> valid

### CODE REVIEW

**Review by odeimaiz — APPROVED** [review]

**Review by Surfict — APPROVED** [review]

**Review by pcrespov — APPROVED** [review]
pair reviewed. Expected test as agreed. thx

File: services/director-v2/tests/unit/test_core_settings.py [hunk 1]
```diff
@@ -157,6 +157,11 @@ def test_expected_failure_dynamic_sidecar_settings(
         ('["two!=no"]', ["two!=no"]),
         ('["one==yes", "two!=no"]', ["one==yes", "two!=no"]),
         ('["     strips.white.spaces   ==  ok "]', ["strips.white.spaces   ==  ok"]),
+        (
+            # Bug from https://github.com/ITISFoundation/osparc-simcore/pull/3190
```
Comment (GitHK) [inline services/director-v2/tests/unit/test_core_settings.py:161]: you can remove this comment, it's fine like this

File: services/director-v2/src/simcore_service_director_v2/core/settings.py [hunk 2]
```diff
@@ -67,7 +67,7 @@
 SUPPORTED_TRAEFIK_LOG_LEVELS: set[str] = {"info", "debug", "warn", "error"}
 
 PlacementConstraintStr = constr(
-    strip_whitespace=True, regex=r"^[a-zA-Z0-9. ]*(!=|==){1}[a-zA-Z0-9. ]*$"
+    strip_whitespace=True, regex=r"^[a-zA-Z0-9_. ]*(!=|==){1}[a-zA-Z0-9. ]*$"
```
Comment (GitHK) [inline services/director-v2/src/simcore_service_director_v2/core/settings.py:70]: may I suggest it like this?
```suggestion
    strip_whitespace=True, regex=r"^[a-zA-Z0-9_. ]*(!=|==){1}[a-zA-Z0-9_. ]*$"
```

File: services/director-v2/tests/unit/test_core_settings.py [hunk 3]
```diff
@@ -157,6 +157,11 @@ def test_expected_failure_dynamic_sidecar_settings(
...
+            '["node.labels.standard_worker==true"]',
+            ["node.labels.standard_worker==true"],
```
Comment (GitHK) [inline services/director-v2/tests/unit/test_core_settings.py:163]: Also add the following test case
```python
('["underscore_label==underscore_value"]', ["underscore_label==underscore_value"]),
```

**Review by GitHK — CHANGES_REQUESTED** [review]

**Comment by mrnicegyu11** [comment]
> Since you are changing these, please also add the above suggestions. Thanks!

I intentionally did not, it is unclear and untested if this is even permitted from docker.
Let's keep the changes small and atomic and not overdo it. 
This change currently prevents master from booting, so I would be happy if it gets in @GitHK

**Comment by GitHK** [comment]
> > Since you are changing these, please also add the above suggestions. Thanks!
> 
> I intentionally did not, it is unclear and untested if this is even permitted from docker. Let's keep the changes small and atomic and not overdo it. This change currently prevents master from booting, so I would be happy if it gets in @GitHK

Let's have a look [at the docs](https://docs.docker.com/config/labels-custom-metadata/#label-keys-and-values). 

Apparently the key part of the label cannot contain underscores but only alphanumerical character and `.` and `-`. Please replace the `_` with `-`  in the tests and in the regex.

Regarding the value part of the label, all valid json encodable characters are ok. So `_` is ok as well.

File: services/director-v2/tests/unit/test_core_settings.py (hunk 1 above)
Comment (GitHK) [reply services/director-v2/tests/unit/test_core_settings.py:161]: just make sure to remove this comment

**Review by GitHK — APPROVED** [review]
👍 Thanks for taking care of this

**Pull request merged**